CalibCalorimetry/HcalAlgos/src/HcalDbHardcode.cc

```diff
@@ -424,19 +424,21 @@ HcalTimingParam HcalDbHardcode::makeTimingParam (HcalGenericDetId fId) {
 #define EMAP_NHTRSHO 4
 #define EMAP_NHSETSHO 3
 
-void HcalDbHardcode::makeHardcodeDcsMap(HcalDcsMap& dcs_map) {
-  dcs_map.mapGeomId2DcsId(HcalDetId(HcalBarrel, -16, 1, 1), 
+std::unique_ptr<HcalDcsMap> HcalDbHardcode::makeHardcodeDcsMap() {
+  HcalDcsMap::Helper dcs_map_helper;
+  dcs_map_helper.mapGeomId2DcsId(HcalDetId(HcalBarrel, -16, 1, 1), 
 			  HcalDcsDetId(HcalDcsBarrel, -1, 1, HcalDcsDetId::HV, 2));
-  dcs_map.mapGeomId2DcsId(HcalDetId(HcalForward, -41, 3, 1), 
+  dcs_map_helper.mapGeomId2DcsId(HcalDetId(HcalForward, -41, 3, 1), 
 			  HcalDcsDetId(HcalDcsForward, -1, 1, HcalDcsDetId::DYN8, 1));
-  dcs_map.mapGeomId2DcsId(HcalDetId(HcalForward, -26, 25, 2), 
+  dcs_map_helper.mapGeomId2DcsId(HcalDetId(HcalForward, -26, 25, 2), 
 			  HcalDcsDetId(HcalDcsForward, -1, 7, HcalDcsDetId::HV, 1));
-  dcs_map.mapGeomId2DcsId(HcalDetId(HcalBarrel, -15, 68, 1), 
+  dcs_map_helper.mapGeomId2DcsId(HcalDetId(HcalBarrel, -15, 68, 1), 
 			  HcalDcsDetId(HcalDcsBarrel, -1, 18, HcalDcsDetId::HV, 3));
-  dcs_map.mapGeomId2DcsId(HcalDetId(HcalOuter, -14, 1, 4), 
+  dcs_map_helper.mapGeomId2DcsId(HcalDetId(HcalOuter, -14, 1, 4), 
 			  HcalDcsDetId(HcalDcsOuter, -2, 2, HcalDcsDetId::HV, 4));
-  dcs_map.mapGeomId2DcsId(HcalDetId(HcalForward, 41, 71, 2), 
+  dcs_map_helper.mapGeomId2DcsId(HcalDetId(HcalForward, 41, 71, 2), 
 			  HcalDcsDetId(HcalDcsForward, 1, 4, HcalDcsDetId::DYN8, 3));
+  return std::make_unique<HcalDcsMap>(dcs_map_helper);
 }
```

Comment:
@franzoni I have no idea, these values have been in the hardcode parameters since before my time (https://github.com/cms-sw/cmssw/blame/CMSSW_6_2_X/CalibCalorimetry/HcalAlgos/src/HcalDbHardcode.cc#L592). This PR is just to improve the code for the objects themselves.